GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Gas.cs

```diff
@@ -0,0 +1,89 @@
+﻿using UnityEngine;
+using UnityEngine.UI;
+
+public class Gas : MonoBehaviour
+{
+    public GameObject noGasText;
+    public GameObject gasLevelImageObj;
+
+    private float _mpg = 0.1f;
+    private float _gasLevel = 5.0f;
+    private float _fullGasLevel = 5.0f;
+    private float _drivedDistanceBeforeLastFill = 0f;
+    private float _totalDistanceDriven = 0f;
```

Comment:
can you explain why you need to keep 2 variables for distance?

Reply:
My calculation of how many gas left is based on distance the car has traveled. Therefore, every time when the player refill the gas, _drivedDistanceBeforeLastFill will be set as the current total distance. And the new _gasLevel will be computed according to the (_totalDistanceDriven -  _drivedDistanceBeforeLastFill), which is the distance the user has driven after filling the gas. 

The reason that I keep two distances instead of just keeping one to record how far the car has driven since last time filled is that I am thinking about that maybe in the future, I will add an indicator to show how far has the car driven in total. Therefore, I may need to keep  _totalDistanceDriven.